opal/include/opal/sys/atomic_stdc.h

```diff
@@ -48,12 +48,20 @@ static inline void opal_atomic_wmb(void)
     atomic_thread_fence(memory_order_release);
 }
 
+/* work around a bug in older gcc versions (observed in gcc 6.x)
+ * where acquire seems to get treated as a no-op instead of being
+ * equivalent to __asm__ __volatile__("": : :"memory") on x86_64
+ * [JS] the issue has still been observed in GCC 7.3.0 but appears
+ *      to be fixed from 8.1.0 onwards
+ */
+#   if  defined(PLATFORM_ARCH_X86_64) &&                        \
+        PLATFORM_COMPILER_GNU && __GNUC__ < 8
+#       define OPAL_ATOMIC_RMB_GCC_FENCE_ACQUIRE_BROKEN 1
```

Comment:
why do we need the extra define?  I'd rather not create another (in this case atomic_stdc-only) define we need to maintain going forward.

Reply:
I thought it'd be cleaner (I had a somewhat more complex condition earlier). I agree, put it back into the code directly.